### PR TITLE
Add bottom bar hit error meter

### LINKED ISSUE #5932: Centered hit errors (like osu!stable)
I can't see the hit errors when it's set to the left or right side. It's further away from the sides than the bottom (Widescreen problem,it's not in the field of view) It would be nice if osu!lazer has the same hit error option as stable, as well as the scaling option.

One Option missed same on the stable:
ss (2019-08-30 at 08 16 00)

**Proposal designs of the feature:** 
Like this (is a screenshot montage, edited by paint.net)
ss (2019-08-30 at 08 29 02)

### PR DESCRIPTION
**Pull request by Game4all**
Closes #5932 

Mostly adapted existing code of the vertical hit bar error meter for an horizontal use

It is for now bottom centered between the song graph and the progress counter
image

### CODE REVIEW

File: osu.Game/Screens/Play/HUD/HitErrorMeters/BottomBarHitErrorMeter.cs [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Scoring;
+using osuTK;
+using osuTK.Graphics;
+using System.Linq;
+
+namespace osu.Game.Screens.Play.HUD.HitErrorMeters
+{
+    public class BottomBarHitErrorMeter : HitErrorMeter
```
Comment (peppy) [inline osu.Game/Screens/Play/HUD/HitErrorMeters/BottomBarHitErrorMeter.cs]: Why is this a complete copy-paste? It should 100% be able to share at least 99% of logic with the side bsrs.

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Play/HUD/HitErrorMeters/BottomBarHitErrorMeter.cs (hunk 1 above)
Comment (Game4all) [reply osu.Game/Screens/Play/HUD/HitErrorMeters/BottomBarHitErrorMeter.cs]: Decided to go with the copy-paste way for the sake of code readability (else code would be filled with inline conditionals) . Will refactor to at least share used components.

**Comment by Game4all** [comment]
Did a bit of refactoring:

- Both bar hit error meters inherit of `BarHitErrorMeter` 
which wraps the judgement, and bar creation from hitwindows.

- Bar hit error meters code is now mostly layout code.

- Bar hit error meters have to implement `CreateJudgement(JudgementResult)` and `CreateColourBar(Color4, HitWindow, bool)` 

- JudgementLine is now public and takes a bool indicating whether it should be vertically displayed or not.

File: osu.Game/Screens/Play/HUD/HitErrorMeters/Bar/SidebarHitErrorMeter.cs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Scoring;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Screens.Play.HUD.HitErrorMeters.Bar
+{
+    public class SidebarHitErrorMeter : BarHitErrorMeter
```
Comment (iiSaLMaN) [inline osu.Game/Screens/Play/HUD/HitErrorMeters/Bar/SidebarHitErrorMeter.cs]: matching the capitalization with the inherited name
```suggestion
    public class SideBarHitErrorMeter : BarHitErrorMeter
```

**Review by frenzibyte — COMMENTED** [review]
I would honestly prefer checks to pick the `Anchor` and `Origin` and whether to set `Width` or `Height`, but I guess this is all for readability

**Comment by smoogipoo** [comment]
You've still duplicated 70% of the code between the vertical and horizontal ones. All the constants are the same, the way in which the gradient is generated is the same...

The only thing that changes is the layout, and that's a matter of correct anchors/origins and positionings.

**Comment by Game4all** [comment]
Will reopen a PR using the method as proposed above (using checks to determine anchors & origin).

**Pull request closed without merging**